enterprise/frontend/src/metabase-enterprise/google_sheets/GoogleSheetManagement.tsx

```diff
@@ -241,10 +241,15 @@ function GoogleSheetsConnectModal({
     event.preventDefault();
     setErrorMessage("");
 
-    const validationRegex = /(https|http)\:\/\/drive\.google\.com\/.+/;
-
-    if (!validationRegex.test(folderLink.trim())) {
-      setErrorMessage(t`Invalid Google Drive folder link`);
+    const folderValidationRegex = /(https|http)\:\/\/drive\.google\.com\/.+/;
+    const sheetValidationRegex =
+      /(https|http)\:\/\/docs\.google\.com\/spreadsheets\/.+/;
```

Comment:
I notice the frontend regex allows `http`, but the backend checks want `https`. If the sync works with http, I think we should let it through in `metabase_enterprise.gsheets.api/url-type`.